File: solutions/python/pangram/2/pangram.py
def is_pangram(sentence):

    letters = set(l for l in sentence.lower() if l.isalpha())
    return len(letters) == 26
    
    # letters = set()
    # for l in sentence:
    #     if l.isalpha():
    #         letters.add(l.lower())
    # return len(letters) == 26

File: solutions/haskell/pangram/1/src/Pangram.hs
module Pangram (isPangram) where
import qualified Data.Set as Set
import Data.Char (toLower)

isPangram :: String -> Bool
isPangram [] = False
isPangram text = my_set == to_compare
  where
    to_compare = Set.fromList "abcdefghijklmnopqrstuvwxyz"
    lower = map toLower text
    filtered = filter (>= 'a') lower
    filtered_again = filter (<= 'z') filtered
    my_set = Set.fromList filtered_again
